### PR TITLE
feat: update registers

### LINKED ISSUE #21: Implement `updateRegisters`

### PR DESCRIPTION
**Pull request by greged93**
Implement the update register function, allowing to update the PC, AP and FP. Closes #21

### CODE REVIEW

File: src/vm/virtualMachine.ts [hunk 1]
```diff
@@ -304,7 +315,206 @@ export class VirtualMachine {
...
+      // If the pc update logic is jump rel, then we add the result
+      // to the pc.
+      case PcUpdate.JumpRel:
+        if (operands.res === undefined) {
+          return {
+            value: undefined,
+            error: new VirtualMachineError(UnconstrainedResError),
+          };
+        }
+        // We check that res in a felt
```
Comment (Eikix) [inline src/vm/virtualMachine.ts:376]: typo?

File: src/vm/virtualMachine.ts [hunk 2]
```diff
@@ -304,7 +315,206 @@ export class VirtualMachine {
...
+        const { value: apPlus2, error: apPlus2Error } = this.runContext.ap.add(
+          UnsignedInteger.TWO_UINT32
+        );
+        if (apPlus2Error !== undefined) {
+          return { value: undefined, error: apPlus2Error };
+        }
+        this.runContext.fp = apPlus2;
+        break;
+      // If the fp update logic is dst, then we add the destination
+      // to the fp is dst if a felt, or set the fp to the destination
```
Comment (Eikix) [inline src/vm/virtualMachine.ts:446]: typo?

**Review by Eikix — APPROVED** [review]
a few typos in comment afai can see
but overall lgtm, 
tests from LC give us confidence to move forward

hopefully we can test further edge cases with fuzzing? ;D
